### PR TITLE
fix: use bang operator for pagination input/output

### PR DESCRIPTION
**Pull request by trivikr**
*Issue #, if available:*
Fixes CI failure in new clients generation https://github.com/aws/aws-sdk-js-v3/pull/1580#issuecomment-713802283

*Description of changes:*
use optional chaining for pagination input/output

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by trivikr** [comment]
Verified: CI is successful for new clients generation https://github.com/aws/aws-sdk-js-v3/pull/1580

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/PaginationGenerator.java [hunk 1]
```diff
@@ -123,7 +123,7 @@ static void generateServicePaginationInterfaces(
 
      private String destructurePath(String path) {
         String[] splitIndex = path.split("\\.");
-        return "['" + String.join("']['", splitIndex) + "']";
+        return "." + String.join("?.", splitIndex);
```
Comment (alexforsyth) [inline smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/PaginationGenerator.java:126]:
```suggestion
        return "."  + path.replace(".", "?.");
```

**Review by alexforsyth — APPROVED** [review]
Im not able to repro `Object is possibly 'undefined'`. I also remember that we indexed in with `[ ... ]` for a reason. Although I don't exactly remember why.

LGTM +1

File: smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/PaginationGenerator.java (hunk 1 above)
Comment (trivikr) [reply smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/PaginationGenerator.java:126]: Findings from offline meeting: The bang operator is used as optional chaining can't be used on the left hand side.

TS Playground: https://www.typescriptlang.org/play?#code/C4TwDgpgBAglC8UDeAoKUCGB+AXFAQigL4DcKKokBCyaUARngM7ABOAlgHYDmx5AxgHtOLKIPoArPHERJSKcRIB02JfRoAifABkYACQ0kgA

**Comment by trivikr** [comment]
Verified: Ci is successful for using bang operator instead of optional chaining in new clients https://github.com/aws/aws-sdk-js-v3/pull/1580

**Pull request merged**